Commonality Analysis Code.Rmd

```diff
@@ -0,0 +1,258 @@
+    across(starts_with("b_dem_rom"), factor),
+    across(starts_with("b_dem_race"), factor))
+
+#Since missForest can only work with categorical variables with a maximum of 53 levels, need to lump together factors. Decided to lump those with less than 10 responses.
+str(cope_ca_data)
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_sex = fct_lump_min(b_dem_sex, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_sex_3_text = fct_lump_min(b_dem_sex_3_text, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_gender = fct_lump_min(b_dem_gender, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_gender_16_text = fct_lump_min(b_dem_gender_16_text, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_orientation = fct_lump_min(b_dem_orientation, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_orientation_7_text = fct_lump_min(b_dem_orientation_7_text, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_sex_experience = fct_lump_min(b_dem_sex_experience, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_sex_experience_10_text = fct_lump_min(b_dem_sex_experience_10_text, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_sex_attract = fct_lump_min(b_dem_sex_attract, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_sex_attract_10_text = fct_lump_min(b_dem_sex_attract_10_text, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_rom_attract = fct_lump_min(b_dem_rom_attract, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_rom_attract_10_text = fct_lump_min(b_dem_rom_attract_10_text, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_race = fct_lump_min(b_dem_race, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_race_7_text = fct_lump_min(b_dem_race_7_text, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_language = fct_lump_min(b_dem_language, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_language_10_text = fct_lump_min(b_dem_language_10_text, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_zip_code = fct_lump_min(b_dem_zip_code, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_grade = fct_lump_min(b_dem_grade, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_grade_9_text = fct_lump_min(b_dem_grade_9_text, 10)) 
+str(cope_ca_data)
+```
+
+```{r data imputation}
+set.seed(12345) #use the same seed number each time to make the code reproducible (the seed number is just the starting point used in the generation of a sequence of random numbers in the random forest)
+
+cope_ca_data_imputed <- missForest(cope_ca_data)
+#NRMSE = 0.25. This is the error term for the continuous variables.
+#PFC = 0.04. This is the error term for the categorical variables.
+#We will run the analyses again with complete cases only (no imputation) as a robustness check to make sure there isn't a huge difference in results.
+```
+
+```{r merge imputed data}
+#Turn the imputed data into it's own object
+cope_ca_data_imputed <- as.data.frame(cope_ca_data_imputed$ximp)
+

```

Comment:
Yes, the imputed dataset does include demographics and baseline data. The reason I did all of these steps is because I didn't want to "make up" data that we did not actually want to impute (data that was only included in the imputation to give missForest information). Our goal with the imputation was to get estimates for missing post-SSI and follow-up outcome data - that's why I extracted these values. I was worried that if I didn't do this and just used the full imputed dataset that we would be including imputed demographics and baseline variables that we don't actually intend to impute (missing data for these values would just stay missing data - only relevant for a small handful of teens). I think I see what you mean since what I'm saying is a non-issue if we drop the teens with missing baseline data to begin with (we don't need to drop the teens with missing demographic data since that doesn't matter for the CA anyways). I added this step in the data cleaning code chunk and got rid of these weird sections. Let me know what you think!

Reply:
Ok yes that makes sense!